mlflow/protos/service.proto

```diff
@@ -351,18 +362,33 @@ message ListExperiments {
 }
 
 message GetExperiment {
-  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+    option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
 
-  // Identifier to get an experiment
-  optional int64 experiment_id = 1 [(validate_required) = true];
+    // Identifier to get an experiment
+    optional int64 experiment_id = 1 [(validate_required) = true];
 
-  message Response {
-    // Returns experiment details
-    optional Experiment experiment = 1;
+    message Response {
+        // Returns experiment details
+        optional Experiment experiment = 1;
 
-    // All (max limit to be imposed) runs associated with this experiment
-    repeated RunInfo runs = 2;
-  }
+        // All (max limit to be imposed) runs associated with this experiment
+        repeated RunInfo runs = 2;
+    }
+}
+
+message GetExperimentByName {
+    option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+    // Human readable name that identifies the experiment.
+    optional string name = 1 [(validate_required) = true];
+
+    message Response {
```

Comment:
Can we reuse the `GetExperiment.Response` object?

Reply:
I have tried couple of heuristics  and none of them seemed to work.  I  have tried the way it was proposed in the link you've provided:
https://developers.google.com/protocol-buffers/docs/proto#nested - didn't work so far.
It seems that GetExperimentByName explicitly requires to have a nested message object Response. I am not sure whether it is possible to do. Maybe you have some ideas?